aws/resource_aws_efs_file_system.go

```diff
@@ -324,6 +362,25 @@ func resourceAwsEfsFileSystemRead(d *schema.ResourceData, meta interface{}) erro
 		return err
 	}
 
+	var policyRes *efs.DescribeFileSystemPolicyOutput
+	policyRes, err = conn.DescribeFileSystemPolicy(&efs.DescribeFileSystemPolicyInput{
+		FileSystemId: fs.FileSystemId,
+	})
+	if err != nil {
+		if isAWSErr(err, efs.ErrCodePolicyNotFound, "") {
+			d.Set("policy", "")
+		} else {
+			return fmt.Errorf("Error describing policy for EFS file system (%s): %s",
+				aws.StringValue(fs.FileSystemId), err)
+		}
+	} else {
+		correctedPolicy := strings.Replace(aws.StringValue(policyRes.Policy),
+			fmt.Sprintf("\"Resource\" : \"%s\",", fsARN), "", 1)
+		if err := d.Set("policy", correctedPolicy); err != nil {
+			return err
+		}
```

Comment:
returned policy contains a "Resource" element which does not exist on create time